alica_engine/include/engine/model/Placeholder.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "AlicaElement.h"
+#include <memory>
+
+namespace alica
+{
+
+class BlackboardBlueprint;
+class PlaceholderFactory;
+
+class Placeholder : public AlicaElement
+{
+public:
+    const BlackboardBlueprint* getBlackboardBlueprint() const { return _blackboardBlueprint.get(); }
+
+private:
+    friend PlaceholderFactory;
+    /**
+     * If nullptr, it will simply receive a reference to its parents Blackboard
```

Comment:
I think we should never allow for placeholders taking by reference, (i.e. placeholders should not allow inheriting blackboard)

Reply:
This seems like a bug actually in the engine. It does not distinguish between an empty blackboard & inherited blackboard. I'll fix

Reply:
this was a misunderstanding on my part of the code. Basically the comment here is incorrect, because createEmpty() is used to create an empty blackboard & it will never be null. I updated the comment